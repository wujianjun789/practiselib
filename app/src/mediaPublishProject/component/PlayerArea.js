import React from 'react';
import PanelFooter from './../../components/PanelFooter';

const PlayerArea = (props) => {
  const footer = <PanelFooter funcNames={['onCancel', 'onConfirm']} btnTitles={['button.cancel', 'button.confirm']}
    btnClassName={['btn-default', 'btn-primary']}
    // btnDisabled={[false, valid]}
    // onCancel={this.onCancel} onConfirm={this.onConfirm}
  />;
  
  return (
    <div id="player-area" className="pro-container">
      <div className="row">
        <div className="form-group">
          <label className="control-label">名称</label>
          <div className="input-container input-w-1">
            <input type="text" className="form-control" />
            <span className="prompt hidden">
              <span>请输入正确参数</span>
            </span>
          </div>
        </div>
      </div>

      <div className="row">
        <div className="form-group">
          <label className="control-label">宽度</label>
          <div className="input-container input-w-1">
            <input type="text" className="form-control" />
            <span className="prompt hidden">
              <span>请输入正确参数</span>
            </span>
          </div>
        </div>
      </div>

      <div className="row">
        <div className="form-group">
          <label className="control-label">高度</label>
          <div className="input-container input-w-1">
            <input type="text" className="form-control" />
            <span className="prompt hidden">
              <span>请输入正确参数</span>
            </span>
          </div>
        </div>
      </div>

      <div className="row">
        <div className="form-group">
          <label className="control-label">X</label>
          <div className="input-container input-w-1">
            <input type="text" className="form-control" />
            <span className="prompt hidden">
              <span>请输入正确参数</span>
            </span>
          </div>
        </div>
      </div>
    
      <div className="row">
        <div className="form-group">
          <label className="control-label">Y</label>
          <div className="input-container input-w-1">
            <input type="text" className="form-control" />
            <span className="prompt hidden">
              <span>请输入正确参数</span>
            </span>
          </div>
        </div>
      </div>

      <div className="row">
        <div className="form-group">
          <label className="control-label">结束</label>
          <div className="input-container input-w-1">
            <select className="form-control" >
              <option>测试一</option>
              <option>测试一</option>
              <option>测试一</option>
            </select>
            <span className="prompt hidden">
              <span>请输入正确参数</span>
            </span>
          </div>
        </div>
      </div>
      {footer}
    </div>
  );
};

export default PlayerArea;