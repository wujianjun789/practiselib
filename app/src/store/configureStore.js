import { createStore, applyMiddleware, compose } from 'redux';
import rootReducer from '../common/reducer';
import thunk from 'redux-thunk';


export default function configureStore(initialState) {

    const composeEnhancers =
        process.env.NODE_ENV !== 'production' &&
            typeof window === 'object' &&
            window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ ?
            window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
                // Specify here name, actionsBlacklist, actionsCreators and other options
            }) : compose;

    const enhancer = composeEnhancers(
        applyMiddleware(thunk),
        // other store enhancers if any
    );
    const store = createStore(rootReducer, initialState, enhancer);
    return store;

}
