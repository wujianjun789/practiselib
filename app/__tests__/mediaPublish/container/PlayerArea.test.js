/**
 * Created by a on 2017/10/23.
 */
import React from 'react';
import {mount} from 'enzyme';
import {Provider} from 'react-redux';
import {IntlProvider} from 'react-intl-redux';
import configureStore from '../../../src/store/configureStore';


import PlayerArea from '../../../src/mediaPublish/container/PlayerArea';


const store = configureStore();
it('PlayerArea renders', ()=>{
    const root = mount(
        <Provider store={store}>
            <IntlProvider>
                <PlayerArea />
            </IntlProvider>
        </Provider>
    );

    expect(root.find('.player-area').length).toEqual(1);
})