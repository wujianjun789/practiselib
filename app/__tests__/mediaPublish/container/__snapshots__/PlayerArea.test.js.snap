exports[`test PlayerArea renders 1`] = `
<div
  className="content player-area">
  <div
    className="left col-sm-3">
    <select
      className="select player-list form-control"
      id=""
      onChange={[Function]}
      placeholder={undefined}
      value="播放表1">
      <option
        value="播放表1">
        播放表1
      </option>
      <option
        value="播放表2">
        播放表2
      </option>
    </select>
    <select
      className="select scene form-control"
      id=""
      onChange={[Function]}
      placeholder={undefined}
      value="场景1">
      <option
        value="场景1">
        场景1
      </option>
      <option
        value="场景2">
        场景2
      </option>
    </select>
    <ul>
      <li>
        区域1
      </li>
      <li>
        区域2
      </li>
    </ul>
    <button
      className="btn btn-primary add-area"
      onClick={[Function]}>
      添加区域
    </button>
  </div>
  <div
    className="center col-sm-3">
    <div
      className="col-sm-8 img-container">
      <img
        src="" />
    </div>
    <div
      className="col-sm-4 pro-container">
      <button
        className="btn btn-primary play"
        onClick={[Function]}>
        播放
      </button>
      <button
        className="btn btn-primary save"
        onClick={[Function]}>
        存为模板
      </button>
      <button
        className="btn btn-primary save-plan"
        onClick={[Function]}>
        保存计划
      </button>
      <button
        className="btn btn-primary quit"
        onClick={[Function]}>
        退出
      </button>
    </div>
  </div>
  <div
    className="right col-sm-6">
    <div
      className="pro-title">
      属性
    </div>
    <div
      className="pro-container">
      <div
        className="form-group row area-name">
        <label
          className="col-sm-3 control-label"
          htmlFor="areaName">
          区域名称：
        </label>
        <div
          className="col-sm-9">
          <input
            className="form-control"
            maxLength="16"
            onChange={[Function]}
            placeholder="输入区域名称"
            type="text"
            value="" />
          <span
            className="prompt hidden">
            仅能使用字母、数字或下划线
          </span>
        </div>
      </div>
      <div
        className="row">
        <div
          className="form-group col-sm-6">
          <label
            className="col-sm-3 control-label"
            htmlFor="width">
            宽度：
          </label>
          <div
            className="col-sm-9">
            <input
              className="form-control "
              maxLength="8"
              onChange={[Function]}
              placeholder="输入宽度"
              type="text"
              value="" />
            <span
              className="prompt hidden">
              仅能使用字母、数字或下划线
            </span>
          </div>
        </div>
        <div
          className="form-group col-sm-6">
          <label
            className="col-sm-3 control-label"
            htmlFor="height">
            高度：
          </label>
          <div
            className="col-sm-9">
            <input
              className="form-control "
              maxLength="8"
              onChange={[Function]}
              placeholder="输入高度"
              type="text"
              value="" />
            <span
              className="prompt hidden">
              仅能使用字母、数字或下划线
            </span>
          </div>
        </div>
      </div>
      <div
        className="row">
        <div
          className="form-group col-sm-6">
          <label
            className="col-sm-3 control-label"
            htmlFor="axisX">
            X轴：
          </label>
          <div
            className="col-sm-9">
            <input
              className="form-control "
              maxLength="8"
              onChange={[Function]}
              placeholder="输入X轴"
              type="text"
              value="" />
            <span
              className="prompt hidden">
              仅能使用字母、数字或下划线
            </span>
          </div>
        </div>
        <div
          className="form-group col-sm-6">
          <label
            className="col-sm-3 control-label"
            htmlFor="axisY">
            Y轴：
          </label>
          <div
            className="col-sm-9">
            <input
              className="form-control "
              maxLength="8"
              onChange={[Function]}
              placeholder="输入Y轴"
              type="text"
              value="" />
            <span
              className="prompt hidden">
              仅能使用字母、数字或下划线
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="asset-lib">
      素材库
    </div>
    <div
      className="asset-container">
      <div
        className="top">
        <select
          className="select asset-type form-control"
          id=""
          onChange={[Function]}
          placeholder={undefined}
          value="类别1">
          <option
            value="类别1">
            类别1
          </option>
          <option
            value="类别2">
            类别2
          </option>
        </select>
        <select
          className="select asset-sort form-control"
          id=""
          onChange={[Function]}
          placeholder={undefined}
          value="素材文字">
          <option
            value="素材文字">
            素材文字
          </option>
          <option
            value="素材图片">
            素材图片
          </option>
        </select>
        <div
          className="searchText asset-search"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}>
          <input
            className="form-control"
            onChange={[Function]}
            placeholder="输入素材名称"
            type="search"
            value="" />
          <ul
            className="" />
          <span
            className="glyphicon glyphicon-search"
            onClick={[Function]} />
        </div>
      </div>
      <div />
    </div>
  </div>
</div>
`;
